### DISCUSSION TITLE
Nation3 Court V1 Proposal [DRAFT]

### DISCUSSION THREAD

**Discussion by xPi2** [nation3/court#16] [Specs]
# Nation3 Court System V1

Original rational by @luisivan: https://forum.nation3.org/t/nation3-court-a-backbone-to-bootstrap-a-circular-economy-by-increasing-trust-between-citizens/553

## Goals & Scope

The goal of the Nation3 Court project it's to provide a system that allow the creation of colateral agreements between two or more non-trusted parties that will be arbitrated by a trusted third party in the case of a dispute.

The scope of this project is to research, design & develop the set of tools required to bootstrap an initial version of this system. At the end of this project we expect to have:
- A Solidity framework to create & manage arbitrated agreements.
- The tools required to setup and operate a court of jurors to arbitrate those agreements.
- An initial human-friendly UI to create & manage agreements, as well as, to resolve disputes.

## Expected Deliverables

- Agreement framework contract.
- Modules to operate the court as a modified multisig. Probably based on Zodiac architecture.
- Subgraph to retrieve agreements information. Including the setup of an indexer Graph node.
- DApp to interact with the system.
- Docs on the system architecture and usage.
- Support to integrate the system with Nation3 setup.

## Team

Coordinator: @xPi2

Estimated to have a 2-5 size team composed of:
- 1-2 Smart contract developers*
- 1-2 Frontend developers*
- 1 UX / UI designer

*(Full-stack developers could fill the gaps also)

## Budget request

The budget requested will account for:

- Compensation for a team of 2-4 developers + 1 designer. (>85%)
- Bounties for outsourced work. (<10%)
- Deployment & setup costs. (<5%)
- Team operational costs.

*It won't account for audit costs.

I propose a budget allocation of 60K USDC for the full project.
In the case of USDC available on the treasury I propose a split of 40K USDC / 20K (USD) in NATION (Valuation at the moment of project completion).

## Timeline estimation

It will be very dependant of the number of contributors to join and the final team capacity.
I would expect 1 month of work in an ideal scenario that could extends to 2 months with less resources.

**Comment by gregfromstl** [comment]
Should we include any sort of timeline estimate?

**Reply by xPi2** [reply]
I would say 1 month for a 4 developers team but I would like to read other opinions on that.

**Comment by xPi2** [comment]
I've updated the proposal based on @gregfromstl & @luisivan feedback:
- Added a rough timeline estimation. 
- Accommodate UX / UI design work to the team instead of relaying on the Design Guild.
- Extend proposal for budget with USDC/NATION split proposals.
